storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1382,5 +1422,438 @@ public static TopologyInfo getTopologyInfo(String name, String asUser, Map storm
+    }
+
+    /* TODO: This function was originally written to replace map-val in util.clj. But we decided to change the coding
+         style in the caller functions to a more Java style for loop. This is mentioned in TODOs across the clojure
+         files.
+    */
+//    public static Map filterVal(IPredicate aFn, Map amap) {
+//        Map newMap = new HashMap();
+//        if (amap == null) {
+//            return newMap;
+//        }
+//        for (Object key: amap.keySet()) {
+//            Object value = amap.get(key);
+//            if(aFn.test(value)) {
+//                newMap.put(key, value);
+//            }
+//        }
+//        return newMap;
+//    }
+
+    /* TODO: This function was originally written to replace filter-key in util.clj. But we decided to change the coding
+         style in the caller functions to a more Java style for loop + if conditionals. This is mentioned in TODOs
+         across the clojure files.
+    */
+//    public static Map filterKey(IPredicate aFn, Map amap) {
+//        Map newMap = new HashMap();
+//        if (amap == null) {
+//            return newMap;
+//        }
+//        for (Object key: amap.keySet()) {
+//            Object value = amap.get(key);
+//            if(aFn.test(key)) {
+//                newMap.put(key, value);
+//            }
+//        }
+//        return newMap;
+//    }
+
+    /* TODO: This function was originally written to replace map-key in util.clj. But we decided to change the coding
+         style in the caller functions to a more Java style for loop. This is mentioned in TODOs across the clojure
+         files.
+    */
+//    public static Map mapKey (IFn aFn, Map amap) {
+//        Map newMap = new HashMap();
+//        if (amap == null) {
+//            return newMap;
+//        }
+//        for (Object key: amap.keySet()) {
+//            Object value = amap.get(key);
+//            Object newKey = aFn.eval(key);
+//            newMap.put(newKey, value);
+//        }
+//        return newMap;
+//    }
+
+    /* TODO: This function was originally written to replace separate in util.clj. But since it was only used in
+    transactional_test.clj, the separate function was moved there for now.
+    */
+    public static Vector<Collection> separate (IPredicate pred, Collection aseq) {
+        Vector<Collection> outputVector = new Vector<Collection>();
+        Collection pass = new HashSet();
+        Collection notPass = new HashSet();
+        for (Object obj: aseq) {
+            if (pred.test(obj)) {
+                pass.add(obj);
+            } else {
+                notPass.add(obj);
+            }
+        }
+        outputVector.add(pass);
+        outputVector.add(notPass);
+        return outputVector;
+    }
+
+    public static void exitProcess (int val, Object... msg) {
+        String combinedErrorMessage = "";
+        for (Object oneMessage: msg) {
+            combinedErrorMessage = combinedErrorMessage + oneMessage.toString();
```

Comment:
Please don't use toString, use String Builder.append.  I know that if it is null it will output "null" but if this is null we will get an NPE.